From ab9433ee42d21c42f596b6e8ea423c6d8815ae33 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:30 -0400
Subject: [ethernet] i40evf: fix off-by-one

Message-id: <1409663591-32531-108-git-send-email-sassmann@redhat.com>
Patchwork-id: 90579
O-Subject: [RHEL7.1 PATCH 107/148] i40evf: fix off-by-one
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

The loop in i40evf_get_rxfh_indir was only reading fifteen registers,
not all sixteen. Change the matching loop in i40evf_set_rxfh_indir at
the same time to make the code more consistent.

Change-ID: I6c182287698e742d1f6ca1a4bcc43cc08df6e1de
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit cc70b080ea773d7aea4a0440316f30f5af6bb655)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c b/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
index 0a5810a..989dbe4 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
@@ -631,7 +631,7 @@ static int i40evf_get_rxfh_indir(struct net_device *netdev, u32 *indir)
  u32 hlut_val;
  int i, j;
 
- for (i = 0, j = 0; i < I40E_VFQF_HLUT_MAX_INDEX; i++) {
+ for (i = 0, j = 0; i <= I40E_VFQF_HLUT_MAX_INDEX; i++) {
   hlut_val = rd32(hw, I40E_VFQF_HLUT(i));
   indir[j++] = hlut_val & 0xff;
   indir[j++] = (hlut_val >> 8) & 0xff;
@@ -656,7 +656,7 @@ static int i40evf_set_rxfh_indir(struct net_device *netdev, const u32 *indir)
  u32 hlut_val;
  int i, j;
 
- for (i = 0, j = 0; i < I40E_VFQF_HLUT_MAX_INDEX + 1; i++) {
+ for (i = 0, j = 0; i <= I40E_VFQF_HLUT_MAX_INDEX; i++) {
   hlut_val = indir[j++];
   hlut_val |= indir[j++] << 8;
   hlut_val |= indir[j++] << 16;
-- 
1.7.1